app/models/mvi.rb

```diff
@@ -111,6 +111,8 @@ def response_from_redis_or_service
     do_cached_with(key: @user.uuid) do
       mvi_service.find_profile(@user)
     end
+  rescue MVI::Errors::RecordNotFound, MVI::Errors::DuplicateRecord
```

Comment:
Discussion around not subclassing RecordNotFound, and instead just explicitly catching the DuplicateRecord

Reply:
Did you want me to change DuplicateRecord to no longer inherit from RecordNotFound?

Reply:
👍 I think so - to me they feel like two distinct errors - from the perspective of the user they may be the same type of error, but from the system's perspective I think the failure behavior can be pretty different.